consumer/consumererror/combine.go

```diff
@@ -15,39 +15,76 @@
 package consumererror
 
 import (
-	"fmt"
+	"errors"
 	"strings"
 )
 
+type combined []error
+
+var (
+	_ error                                    = (*combined)(nil)
+	_ interface{ Is(target error) bool }       = (*combined)(nil)
+	_ interface{ As(target interface{}) bool } = (*combined)(nil)
+)
+
+func (c combined) Error() string {
+	var sb strings.Builder
+	length := len(c)
+	for i, err := range c {
+		sb.WriteString(err.Error())
+		if i != length-1 {
+			sb.WriteString("; ")
+		}
+	}
+	return "[" + sb.String() + "]"
+}
+
+// Range iterates over the slice and all subslices of errors
+// and will apply the passed function on each.
+// The method will return once a the passed function returns true
+// or there is no items in the slice.
+func (c combined) Range(fn func(error) bool) bool {
+	for _, err := range c {
+		if target, ok := err.(combined); ok {
```

Comment:
I don't think this is necessary, as since errors.Is and errors.As will use Is and As methods, if they are defined by the error: https://pkg.go.dev/errors#As.

Reply:
😲 , I really didn't think about that. Definitely simplifies the logic further.